controllers/cheuser/controller.go

```diff
@@ -0,0 +1,344 @@
+				}
+			}
+		}),
+	}
+}
+
+func (r *CheUserNamespaceReconciler) watchRulesForConfigMaps(ctx context.Context) handler.EventHandler {
+	return &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(mo handler.MapObject) []reconcile.Request {
+			if isLabeledAsUserSettings(mo.Meta) && r.isInManagedNamespace(ctx, mo.Meta) {
+				return asReconcileRequestForNamespace(mo.Meta)
+			} else {
+				return []reconcile.Request{}
+			}
+		}),
+	}
+}
+
+func isLabeledAsUserSettings(obj metav1.Object) bool {
+	return obj.GetLabels()["app.kubernetes.io/component"] == userSettingsComponentLabelValue
+}
+
+func (r *CheUserNamespaceReconciler) isInManagedNamespace(ctx context.Context, obj metav1.Object) bool {
+	info, err := r.namespaceCache.GetNamespaceInfo(ctx, obj.GetNamespace())
+	return err == nil && info != nil && info.OwnerUid != ""
+}
+
+func (r *CheUserNamespaceReconciler) triggerAllNamespaces(ctx context.Context) handler.EventHandler {
+	return &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(mo handler.MapObject) []reconcile.Request {
+			nss := r.namespaceCache.GetAllKnownNamespaces()
+			ret := make([]reconcile.Request, len(nss))
+
+			for _, ns := range nss {
+				ret = append(ret, reconcile.Request{
+					NamespacedName: types.NamespacedName{Name: ns},
+				})
+			}
+
+			return ret
+		}),
+	}
+}
+
+func (r *CheUserNamespaceReconciler) hasCheCluster(ctx context.Context, namespace string) bool {
+	list := v1.CheClusterList{}
+	if err := r.client.List(ctx, &list, client.InNamespace(namespace)); err != nil {
+		return false
+	}
+
+	return len(list.Items) > 0
+}
+
+func asReconcileRequestForNamespace(obj metav1.Object) []reconcile.Request {
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{Name: obj.GetNamespace()},
+		},
+	}
+}
+
+func (r *CheUserNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	info, err := r.namespaceCache.ExamineNamespace(ctx, req.Name)
+	if err != nil {
+		logrus.Errorf("Failed to examine namespace %s for presence of Che user info labels: %v", req.Name, err)
+		return ctrl.Result{}, err
+	}
+
+	if info == nil || info.OwnerUid == "" {
+		// we're not handling this namespace
+		return ctrl.Result{}, nil
+	}
+
+	checluster := findManagingCheCluster(*info.CheCluster)
+	if checluster == nil {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if devworkspace.GetDevworkspaceState(r.scheme, checluster) != devworkspace.DevworkspaceStateEnabled {
+		return ctrl.Result{}, nil
+	}
+
+	// let's construct the deployContext to be able to use methods from v1 operator
+	deployContext := &deploy.DeployContext{
+		CheCluster: org.AsV1(checluster),
+		ClusterAPI: deploy.ClusterAPI{
+			Client:          r.client,
+			NonCachedClient: r.client,
+			DiscoveryClient: nil,
+			Scheme:          r.scheme,
+		},
+	}
+
+	isSelfSignedCertUsed, err := deploy.IsSelfSignedCertificateUsed(deployContext)
```

Comment:
It is better to check if corresponding secret exists without invoking `IsSelfSignedCertificatdUsed`

Reply:
A common problem is that reconciling self-signed certificate is not enough. 
Others certificates are located in `ca-certs-merged` config map
see https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/tls.go#L505

Reply:
Besides this, we have `che-git-self-signed-cert` configmap which store self-signed certificate and host for git repository. see https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/server/server_deployment.go#L84-L109

Reply:
`IsSelfSignedCertificateUsed` is no longer used and `ca-certs-merged` should be handled. I chose not to handle `che-git-self-signed-cert` just to not further increase the scope of this issue. We can handle it separately later (I am also glad, we managed to make the CI (almost) pass after quite some time, so I don't want to go through that ordeal again ;) ).

Reply:
Btw handling of the git cert is more complex, because it only makes sense if we also sync gitconfig - for that we not only need the git host (that should be present in the `che-git-self-signed-cert` configmap, but also user name and email. This is intentionally out of scope of this PR so we'll need to remember to handle that in a separate issue in the future.